src/Orleans.Runtime/GrainDirectory/ILocalClientDirectory.cs

```diff
@@ -0,0 +1,11 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.GrainDirectory
+{
+    internal interface ILocalClientDirectory
+    {
+        bool TryLocalLookup(GrainId grainId, out List<ActivationAddress> addresses);
```

Comment:
Since we have a `ClientGrainId` struct, I think we should use it in this interface

Reply:
The reason I didn't use that is that it's not a "transfer type" - i.e, I don't want it to exist over the wire, only within a single process